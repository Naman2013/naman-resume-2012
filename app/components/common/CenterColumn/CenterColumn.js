import React from 'react';
import PropTypes from 'prop-types';
import style from './CenterColumn.style';

const CenterColumn = ({ children, theme }) => (
  <div className="root" style={theme}>
    {children}

    <style jsx>{style}</style>
  </div>
);

CenterColumn.propTypes = {
  children: PropTypes.node.isRequired,
  theme: PropTypes.shape({}),
};

CenterColumn.defaultProps = {
  theme: {},
};

export default CenterColumn;
