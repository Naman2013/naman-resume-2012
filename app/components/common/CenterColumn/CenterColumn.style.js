import css from 'styled-jsx/css';

export default css`
  .root {
    width: 100%;
    margin: 0 auto;
    transition: width 0.25s;
  }`;
