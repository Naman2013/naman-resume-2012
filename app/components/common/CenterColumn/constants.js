export const defaultWidths = ['620px', '780px', '940px'];
