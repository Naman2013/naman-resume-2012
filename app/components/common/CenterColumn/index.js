import CenterColumn from './CenterColumn';

export default CenterColumn;
