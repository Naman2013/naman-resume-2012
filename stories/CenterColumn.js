import React from 'react';
import { storiesOf } from '@storybook/react';
import CenterColumn from '../app/components/common/CenterColumn';

storiesOf('CenterColumn', module)
  .add('Default', () => (
    <CenterColumn theme={{ backgroundColor: 'aqua' }}>
      <div style={{ padding: '100px', textAlign: 'center' }}>
        <h1>Sample header</h1>
        <p>Sample content</p>
      </div>
    </CenterColumn>
  ));
