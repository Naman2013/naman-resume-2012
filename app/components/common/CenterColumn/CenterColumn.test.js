import React from 'react';
import { shallow } from 'enzyme';
import CenterColumn from './CenterColumn';

describe('CenterColumn', () => {
  const shallowWrapper = shallow(<CenterColumn><div /></CenterColumn>);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
