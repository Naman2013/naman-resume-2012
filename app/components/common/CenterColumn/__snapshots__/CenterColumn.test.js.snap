// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterColumn should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterColumn
    breakpoints={
      Array [
        "screen and (min-width: 768px)",
        "screen and (min-width: 1024px)",
        "screen and (min-width: 1366px)",
      ]
    }
    theme={Object {}}
    widths={
      Array [
        "620px",
        "780px",
        "940px",
      ]
    }
  >
    <div />
  </CenterColumn>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div />,
        <JSXStyle
          css={
            Array [
              ".root.jsx-711544156{width:100%;margin:0 auto;-webkit-transition:width 0.25s;transition:width 0.25s;}",
            ]
          }
          styleId="711544156"
        />,
        <JSXStyle
          css={
            Array [
              "@media screen and (min-width: 768px){.root.__jsx-style-dynamic-selector{width:620px;}}",
              "@media screen and (min-width: 1024px){.root.__jsx-style-dynamic-selector{width:780px;}}",
              "@media screen and (min-width: 1366px){.root.__jsx-style-dynamic-selector{width:940px;}}",
            ]
          }
          dynamic={
            Array [
              "screen and (min-width: 768px)",
              "620px",
              "screen and (min-width: 1024px)",
              "780px",
              "screen and (min-width: 1366px)",
              "940px",
            ]
          }
          styleId="2989590773"
        />,
      ],
      "className": "jsx-711544156 jsx-3002582582 root",
      "style": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".root.jsx-711544156{width:100%;margin:0 auto;-webkit-transition:width 0.25s;transition:width 0.25s;}",
          ],
          "styleId": "711544156",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "@media screen and (min-width: 768px){.root.__jsx-style-dynamic-selector{width:620px;}}",
            "@media screen and (min-width: 1024px){.root.__jsx-style-dynamic-selector{width:780px;}}",
            "@media screen and (min-width: 1366px){.root.__jsx-style-dynamic-selector{width:940px;}}",
          ],
          "dynamic": Array [
            "screen and (min-width: 768px)",
            "620px",
            "screen and (min-width: 1024px)",
            "780px",
            "screen and (min-width: 1366px)",
            "940px",
          ],
          "styleId": "2989590773",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div />,
          <JSXStyle
            css={
              Array [
                ".root.jsx-711544156{width:100%;margin:0 auto;-webkit-transition:width 0.25s;transition:width 0.25s;}",
              ]
            }
            styleId="711544156"
          />,
          <JSXStyle
            css={
              Array [
                "@media screen and (min-width: 768px){.root.__jsx-style-dynamic-selector{width:620px;}}",
                "@media screen and (min-width: 1024px){.root.__jsx-style-dynamic-selector{width:780px;}}",
                "@media screen and (min-width: 1366px){.root.__jsx-style-dynamic-selector{width:940px;}}",
              ]
            }
            dynamic={
              Array [
                "screen and (min-width: 768px)",
                "620px",
                "screen and (min-width: 1024px)",
                "780px",
                "screen and (min-width: 1366px)",
                "940px",
              ]
            }
            styleId="2989590773"
          />,
        ],
        "className": "jsx-711544156 jsx-3002582582 root",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".root.jsx-711544156{width:100%;margin:0 auto;-webkit-transition:width 0.25s;transition:width 0.25s;}",
            ],
            "styleId": "711544156",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "@media screen and (min-width: 768px){.root.__jsx-style-dynamic-selector{width:620px;}}",
              "@media screen and (min-width: 1024px){.root.__jsx-style-dynamic-selector{width:780px;}}",
              "@media screen and (min-width: 1366px){.root.__jsx-style-dynamic-selector{width:940px;}}",
            ],
            "dynamic": Array [
              "screen and (min-width: 768px)",
              "620px",
              "screen and (min-width: 1024px)",
              "780px",
              "screen and (min-width: 1366px)",
              "940px",
            ],
            "styleId": "2989590773",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
